src/client/src/views/SceneForm.js

```diff
@@ -14,7 +14,6 @@ module.exports = {
       m('form.save-name-form', {
         onsubmit: function(e) {
           e.preventDefault()
-          Scene.save()
```

Comment:
I think if we are going to have a "form" element, we should keep the standard/expected behavior of having the submit event trigger the action. What you have done is basically removed any action on the submit event, and instead added a click handler to the button. The standard way to do forms is to have a button fire the submit event for the form, and then have the onsubmit handler of the form execute the action you want.

This is what we had originally. So I would probably want this to go back to what we had.

Reply:
@evanhackett If that's how you want it, then instead of everything calling Scene.save(), each form should have it's own save function in the model. It would also keep the error checking separate.

Reply:
@JosephJindrich I'm not saying you have to do it that way. 

Let's take a step back. I think I need to explain myself more clearly.

So, currently you have two form elements. You disabled both of their onsubmit handlers. For one of them (save-name-form), you also removed the button that would submit the form. So this form not only has a submit event handler that does nothing except disable the default event (line 16), but it also can't even be submitted because you removed the button that would submit it (line 26). So why is save-name-form even its own form at this point? 

Then, for save-start-form, you also have an empty onsubmit handler (aside from preventing the default form submit action) which is on lines 28-30. Once again, how is this form even submitted anyway? You don't have a way to submit the form, because you don't have a submit button. You do have a button that is not tied to the form, but instead executes a click handler that makes it _effectively_ tied to the form. Which brings me back to my original point: why not keep the web standard form behavior that we already demonstrated? Why disable the onsubmit event, remove a submit button, and then add a button with a clickhandler that effectively does what submitting the form would do?

As I mentioned in another comment, using forms with their built-in events is the proper way to do this, because you get some semantic things for free, especially accessibility stuff (like screen readers and submitting forms as a blind individual). Having your own ad-hoc forms by combining input elements with buttons that have click handlers _does technically work_, but the thing is we already put in the work to do it the right way with forms, so there is no reason to undo that work we already did (which is my problem with what your commit does).

Ok, now lets fast-forward to what you said:

> If that's how you want it, then instead of everything calling Scene.save(), each form should have it's own save function in the model. It would also keep the error checking separate.

So, the question here is what do you think I want when you say "that's how you want it"? Because I don't think what I said has anything to do with a specific implementation. I never said "you have to make each thing its own form". If you wanted to only have 1 form, with 1 save function, I'm actually ok with that. But then the html has to actually only have 1 form. Right now there are 2. So if you wanted to go this route that you need to get rid of the save-name-form and instead refactor it so that it is simply an input that is part of the larger form (the save-changes-form).

The other strategy would be to have multiple forms. As you pointed out, maybe the best thing at that point would be to have multiple save functions in the model for each different form. I think we could probably reuse the save function we have, but we would want to wrap it with separate error checking for each separate form.

I think I the idea of having one large form that has multiple inputs for all of the different options. I think that is also what you were going for. So like I said, if you want to go this route just clean up the code so that there is actually only one form (with multiple input elements). 

I hope that clears things up!

Reply:
Sorry for the late reply. I don't mind switching it, and I didn't mean it to be sassy when I said "If that's how you want it" I just meant if you think it's a better way to go we should break apart the save. Since we had a blanket save function, it made the most sense to me to make a blanket button. As the save button didn't only do things inside the form I would have made, it would have done it everywhere. If I'm going to put the button inside the form, I would like to make it so the save function is broken into separate pieces. I'll refactor the code when I have a minute and you can look over the changes I'm thinking.

Reply:
@evanhackett Alright take a look at the new commit. I think it makes it a bit more understandable about what I was talking about.

Reply:
I like the idea of having one form with multiple input fields, so I like the way you have done it here.

I'm not 100% sure I agree with having separate save functions instead of just the one, but I'm ok with it for now. I was originally thinking we would have 1 save function in the model, and the error handling would be done inside the view. But honestly this works for me.

Reply:
The way we had it, when you added a caption it would save anything currently in the text boxes.